Jenkinsfile

```diff
@@ -167,14 +170,33 @@ pipeline {
           environment {
             platform = 'centos6'
           }
-          steps {
-            sh 'rm -f apache-couchdb-*.tar.gz'
-            unstash 'tarball'
-            sh( script: build_script )
-          } // steps
+          stages {
+            stage('Build from tarball & test') {
+              steps {
+                unstash 'tarball'
+                sh( script: build_and_test )
+              }
+              post {
+                always {
+                  junit '**/.eunit/*.xml, **/_build/*/lib/couchdbtest/*.xml'
+                }
+              }
+            }
+            stage('Build CouchDB packages') {
+              steps {
+                sh( script: make_packages )
+                sh( script: cleanup_and_save )
+              }
+              post {
+                success {
+                  archiveArtifacts artifacts: 'pkgs/**', fingerprint: true
+                }
+              }
+            }
+          } // stages
           post {
-            success {
-              archiveArtifacts artifacts: 'pkgs/**', fingerprint: true
+            cleanup {
+              sh 'rm -rf ${WORKSPACE}/*'
```

Comment:
This leaves the actual customWorkspace directory. Are those going to pile up? We can ask Infra to look at our workspaces if necessary. Have you tried an `ls -la ${WORKSPACE}/..` just to see?

Reply:
Ah, you mean the original workspace before the `customWorkspace()` invocation changes it?

The way I observed this working was that the invocation of `customWorkspace()` would change the `${WORKSPACE}` environment variable as well, so this `cleanup` action will remove the contents of e.g. the `centos6` sub-folder of the original workspace. You're right, the original workspace directory will still exist. It won't take up any space. I don't know if that's a problem. I have to believe Jenkins manages workspace reclamations in general, but that's an assumption on my part.